src/rust/src/asn1.rs

```diff
@@ -77,21 +78,42 @@ struct DssSignature<'a> {
     s: asn1::BigUint<'a>,
 }
 
-pub(crate) fn big_asn1_uint_to_py<'p>(
+pub(crate) fn big_byte_slice_to_py_int<'p>(
     py: pyo3::Python<'p>,
-    v: asn1::BigUint<'_>,
+    v: &'_ [u8],
 ) -> pyo3::PyResult<&'p pyo3::PyAny> {
+    let signed = big_byte_slice_is_negative(py, v)?;
```

Comment:
This seems like a weird way to solve this, why not just take `signed` as a parameter that teh caller specifies based on context?